packages/main/src/themes/base/Switch.less

```diff
@@ -0,0 +1,319 @@
+@_ui5_switch_track_hover_bg: darken(@sapUiButtonBackground, 7);
+@_ui5_switch_track_checked_border_color: lighten(desaturate(@sapUiToggleButtonPressedBorderColor, 46), 19);
+@_ui5_switch_track_hover_border_color: @_ui5_switch_track_checked_border_color;
+@_ui5_switch_track_border_radius: 0.75rem;
+
+// switch handle
+@_ui5_switch_handle_width: 2rem;
+@_ui5_switch_handle_height: 2rem;
+@_ui5_switch_handle_border_width: 1px;
+@_ui5_switch_handle_border_radius: 1rem;
+@_ui5_switch_handle_bg: darken(@sapUiButtonBackground, 2);
+@_ui5_switch_handle_hover_bg: darken(@sapUiButtonHoverBackground, 2);
+@_ui5_switch_handle_checked_hover_bg: @sapUiToggleButtonPressedHoverBackground;
+@_ui5_switch_handle_semantic_hover_bg: @sapUiErrorBG;
+@_ui5_switch_handle_semantic_checked_hover_bg: @sapUiSuccessBG;
+@_ui5_switch_handle_semantic_hover_border_color: @sapUiErrorBorder;
+@_ui5_switch_handle_semantic_checked_hover_border_color: @sapUiSuccessBorder;
+@_ui5_switch_handle_compact_width: 1.625rem;
+@_ui5_switch_handle_compact_height: 1.625rem;
+
+:host(ui5-switch) {
+	display: inline-block;
+}
+
+ui5-switch {
+	display: inline-block;
+}
+
+span[data-sap-ui-wc-root] {
+	display: inline-block;
+	width: 100%;
+}
+
+.ui5-switch {
+	position: relative;
+	width: 100%;
+	height: @_ui5_switch_height;
+	min-width: @_ui5_switch_width;
+	cursor: pointer;
+	outline: none;
+	-webkit-tap-highlight-color: rgba(0,0,0,0); // to remove the gray highlight, when tapping active elements in Mobile Safari
+}
+
+.ui5-switch-inner {
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	height: 100%;
+	overflow: hidden;
+	pointer-events: none;
+}
+
+.ui5-switch-track {
+	height: @_ui5_switch_track_height;
+	width: 100%;
+	display: flex;
+	align-items: center;
+	background: @_ui5_switch_track_bg;
+	border: 1px solid;
+	border-color: @sapUiContentForegroundBorderColor;
+	border-radius: @_ui5_switch_track_border_radius;
+	box-sizing: border-box;
+}
+
+.ui5-switch-slider {
+	position: relative;
+	height: @_ui5_switch_height;
+	width: 100%;
+	transition: transform 0.1s ease-in;
+	transform-origin: top left;
+}
+
+.ui5-switch-handle {
+	position: absolute;
+	left: -1px;
+	width: @_ui5_switch_handle_width;
+	height: @_ui5_switch_handle_height;
+	background: @_ui5_switch_handle_bg;
+	border: @_ui5_switch_handle_border_width solid @sapUiContentForegroundBorderColor;
+	border-radius: @_ui5_switch_handle_border_radius; 
+	box-sizing: border-box;
+}
+
+.ui5-switch-text {
+	display: flex;
+	justify-content: center;
+	position: absolute;
+	min-width: 1.625rem;
+	padding: 0 0.125rem;
+	font-size: @sapMFontSmallSize;
+	font-family: "72","72full",Arial,Helvetica,sans-serif;
+	text-transform: uppercase; 
+	text-align: center;
+	color: @sapUiBaseText;
+	white-space: nowrap;
+	user-select: none;
+	-webkit-user-select: none;
+	-ms-user-select: none;
+}
+
+.ui5-switch-text--on {
+	left: calc(~"-100% + 1.9125rem");
```

Comment:
hmm, should you escape the whole `calc` expression like `~"calc(100% .. )"`

Reply:
> hmm, should you escape the whole `calc` expression like `~"calc(100% .. )"`

Yes, in the less files we should escape the calc, once we move the styles in css, we can remove the escaping.